server/services/prisonerSearchService.test.ts

```diff
@@ -0,0 +1,67 @@
+import type { Prisoner } from 'prisonRegisterApiClient'
+import { HmppsAuthClient, PrisonerSearchClient } from '../data'
+import PrisonerSearchService from './prisonerSearchService'
+
+describe('prisonerSearchService', () => {
+  const hmppsAuthClient = {
+    getSystemClientToken: jest.fn(),
+  }
+
+  const prisonerSearchClient = {
+    getPrisonerByPrisonNumber: jest.fn(),
+  }
+
+  const prisonerSearchService = new PrisonerSearchService(
+    hmppsAuthClient as unknown as HmppsAuthClient,
+    prisonerSearchClient as unknown as PrisonerSearchClient,
+  )
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  describe('getPrisonerByPrisonNumber', () => {
+    it('should get prisoner by prison number given prisoner search returns a prisoner', async () => {
+      // Given
+      const prisonNumber = 'A1234BC'
+
+      const systemToken = 'a-system-token'
+      hmppsAuthClient.getSystemClientToken.mockImplementation(() => Promise.resolve(systemToken))
+
+      const prisoner: Prisoner = {
+        prisonerNumber: prisonNumber,
+        firstName: 'AUSTIN',
+        lastName: 'AVARY',
+        dateOfBirth: '1993-12-08',
+        gender: 'Male',
+      }
+      prisonerSearchClient.getPrisonerByPrisonNumber.mockImplementation(() => Promise.resolve(prisoner))
```

Comment:
Just like mockito 👍

Reply:
Not quite as elegant / convenient to use, but yes, conceptually the same 👍